pkg/specgen/specgen.go

```diff
@@ -594,28 +597,37 @@ var (
 )
 
 // NewSpecGenerator returns a SpecGenerator struct given one of two mandatory inputs
-func NewSpecGenerator(arg string, rootfs bool) *SpecGenerator {
+func NewSpecGenerator(arg string, rootfs bool) (*SpecGenerator, error) {
 	csc := ContainerStorageConfig{}
 	if rootfs {
 		csc.Rootfs = arg
 		// check if rootfs should use overlay
-		lastColonIndex := strings.LastIndex(csc.Rootfs, ":")
-		if lastColonIndex != -1 && lastColonIndex+1 < len(csc.Rootfs) && csc.Rootfs[lastColonIndex+1:] == "O" {
-			csc.RootfsOverlay = true
-			csc.Rootfs = csc.Rootfs[:lastColonIndex]
+		parts := strings.SplitN(arg, ":", 2)
+		if len(parts) == 2 {
+			for _, opt := range strings.Split(parts[1], ",") {
+				switch {
+				case (opt == "idmap" || strings.HasPrefix(opt, "idmap=")):
+					csc.RootfsMapping = &opt
+				case opt == "O":
+					csc.RootfsOverlay = true
+				default:
+					return nil, fmt.Errorf("unknown option %q", opt)
+				}
+			}
+			csc.Rootfs = parts[0]
```

Comment:
This will break backwards compat, now it will fail if a path contains a colon. As I understand the old code carefully checked that only `:O` was interpreted as option and everything else considered a regular path.

I don't think we can do this if we a serious about  backwards compat. Should we add a new flag for this?

Reply:
yes I know we break that, but that is a bit weird corner case and we already use `:` as separator for other options (e.g. `--volume`).  The previous code was not that much clearer where we only threated `:O` as an option and everything else as part of the path.  I'd not worry about adding another option just to change this behavior.

Reply:
This is breaking and at least one user depends on it see the linked issue #11913.
For volumes there is `--mount` which allows paths to contain a colon.

Reply:
should we just add another exception for "idmap" and avoid arguments?

@rhatdan what do you think?

Reply:
supporting "idmap" and arguments is easy in the same way we handle "O" but it is a mess when we want to provide both options like `:O,idmap`